### PR TITLE
Wip edit profile

### PR DESCRIPTION
**Pull request by wpflueger**

### CODE REVIEW

File: upmed-api/src/api/patient/patient.py [hunk 1]
```diff
@@ -1,7 +1,294 @@
...
+        email = request.json['email']
+        res = pat.document(str(pid)).get()
+        res = res.to_dict()
+        # print(pid)
+        if res['email'] == email:
+            utype = "PATIENT"
+            auth_token = auth.encode_auth_token(pid, utype)
+            resp = {
+                "googleId": pid,
+                "token": auth_token.decode()
```
Comment (kenkenchuen) [inline upmed-api/src/api/patient/patient.py:32]: token should just be the encoded auth_token, don't decode it

Comment (wpflueger) [reply upmed-api/src/api/patient/patient.py:32]: That is to decode from the byte encoded string to a normal string

File: upmed-api/src/api/patient/patient.py [hunk 2]
```diff
@@ -1,7 +1,294 @@
...
+            "smoker": patient.smoker,
+            "calendar": patient.calendar,
+            "health": patient.health,
+            "doctors": patient.doctors
+
+        })
+        auth_token = auth.encode_auth_token(patient.id, utype)
+        responseObject = {
+            'id': patient.id,
+            'auth_token': auth_token.decode()
```
Comment (kenkenchuen) [inline upmed-api/src/api/patient/patient.py:91]: same as other comment about token

File: upmed-api/src/api/patient/patient.py [hunk 3]
```diff
@@ -1,7 +1,294 @@
...
+        delete() : Delete a document from Firestore collection.
+    """
+    post_data = request.get_json()
+    try:
+        # Check for ID in URL query
+        pid = post_data.get('id')
+        pat.document(pid).delete()
+        return jsonify({"success": True}), 200
+    except Exception as e:
+        return f"An Error Occured: {e}"
```
Comment (kenkenchuen) [inline upmed-api/src/api/patient/patient.py:113]: random f

File: upmed-api/src/api/patient/patient.py (hunk 2 above)
Comment (wpflueger) [reply upmed-api/src/api/patient/patient.py:91]: That is to decode from the byte encoded string to a normal string @kenkenchuen

File: upmed-api/src/api/patient/patient.py (hunk 3 above)
Comment (wpflueger) [reply upmed-api/src/api/patient/patient.py:113]: I have that to have the formatted string to return the error. It is needed for an f-string in python3. @kenkenchuen

File: upmed-api/src/api/patient/patient.py [hunk 4]
```diff
@@ -1,7 +1,294 @@
...
+
+@patient_endpoints.route('/getByToken', methods=['POST'])
+def getbytoken():
```
Comment (kenkenchuen) [inline upmed-api/src/api/patient/patient.py:116]: request body contains the token, no headers

you can just try-except decoding the token

File: upmed-api/src/api/patient/patient.py [hunk 5]
```diff
@@ -1,7 +1,294 @@
...
+        return make_response(jsonify(responseObject)), 200
+    else:
+        responseObject = {
+            'status': 'fail',
+            'message': 'Provide a valid auth token.'
+        }
+        return make_response(jsonify(responseObject)), 401
+
+
+@patient_endpoints.route('/getRecords', methods=['POST'])
```
Comment (kenkenchuen) [inline upmed-api/src/api/patient/patient.py:175]: same as above

File: upmed-api/src/api/patient/patient.py [hunk 6]
```diff
@@ -1,7 +1,294 @@
...
+    else:
+        responseObject = {
+            'status': 'fail',
+            'message': 'Provide a valid auth token.'
+        }
+        return make_response(jsonify(responseObject)), 401
+
+
+@patient_endpoints.route('/editProfile', methods=['POST'])
+def editProfile():
```
Comment (kenkenchuen) [inline upmed-api/src/api/patient/patient.py:230]: same as above

File: upmed-api/src/api/hcp/hcp.py [hunk 7]
```diff
@@ -1,7 +1,573 @@
...
+        email = request.json['email']
+        res = hcpdb.document(str(pid)).get()
+        res = res.to_dict()
+        print(pid)
+        if res['email'] == email:
+            utype = "HCP"
+            auth_token = auth.encode_auth_token(pid, utype)
+            resp = {
+                "googleId": pid,
+                "token": auth_token.decode()
```
Comment (kenkenchuen) [inline upmed-api/src/api/hcp/hcp.py:43]: dont decode token

File: upmed-api/src/api/hcp/hcp.py [hunk 8]
```diff
@@ -1,7 +1,573 @@
...
+            "calendar": hcp.calendar,
+            "specialty": hcp.specialty,
+            "title": hcp.title,
+            "hours": hours,
+            "patients": hcp.patients
+        })
+        auth_token = auth.encode_auth_token(hcp.id, utype)
+        responseObject = {
+            'id': hcp.id,
+            'auth_token': auth_token.decode()
```
Comment (kenkenchuen) [inline upmed-api/src/api/hcp/hcp.py:237]: don't decode token

File: upmed-api/src/api/hcp/hcp.py [hunk 9]
```diff
@@ -1,7 +1,573 @@
...
+        delete() : Delete a document from Firestore collection.
+    """
+    post_data = request.get_json()
+    try:
+        # Check for ID in URL query
+        pid = post_data.get('id')
+        hcpdb.document(pid).delete()
+        return jsonify({"success": True}), 200
+    except Exception as e:
+        return f"An Error Occured: {e}"
```
Comment (kenkenchuen) [inline upmed-api/src/api/hcp/hcp.py:260]: remove f

File: upmed-api/src/api/hcp/hcp.py [hunk 10]
```diff
@@ -1,7 +1,573 @@
...
+
+
+@hcp_endpoints.route('/getByToken', methods=['POST'])
+def getbytoken():
+    # get the auth token
+    auth_header = request.headers.get('Authorization')
```
Comment (kenkenchuen) [inline upmed-api/src/api/hcp/hcp.py:266]: no auth_header, only request only contains token

File: upmed-api/src/api/hcp/hcp.py [hunk 11]
```diff
@@ -1,7 +1,573 @@
...
+
+
+@hcp_endpoints.route('/setRecords', methods=['POST'])
+def set_health_event():
+    """
+    Creates a health event given an HCP token, a patient ID
+    and the criteria of a health event
+    :return: HealthEvent
+    """
+    auth_header = request.headers.get('Authorization')
```
Comment (kenkenchuen) [inline upmed-api/src/api/hcp/hcp.py:405]: same thing about no auth_header

File: upmed-api/src/api/hcp/hcp.py [hunk 12]
```diff
@@ -1,7 +1,573 @@
...
+        responseObject = {
+            'status': 'fail',
+            'message': 'Provide a valid auth token.'
+        }
+        return make_response(jsonify(responseObject)), 401
+
+
+@hcp_endpoints.route('/notify', methods=['POST'])
+def notify():
+    auth_header = request.headers.get('Authorization')
```
Comment (kenkenchuen) [inline upmed-api/src/api/hcp/hcp.py:497]: same thing auth_header

File: upmed-api/src/api/patient/patient.py (hunk 1 above)
Comment (kenkenchuen) [reply upmed-api/src/api/patient/patient.py:32]: we want the encoded string to pass around, front end will decode it itself

File: upmed-api/src/api/patient/patient.py (hunk 2 above)
Comment (kenkenchuen) [reply upmed-api/src/api/patient/patient.py:91]: we want the encoded string to pass around, front end will decode it itself

**Comment by wpflueger** [comment]
@kenkenchuen I have addressed all instances of needing to move the token to the body

**Review by kenkenchuen — APPROVED** [review]

**Pull request merged**